src/google/protobuf/compiler/php/php_generator.h

```diff
@@ -37,7 +37,13 @@ class PROTOC_EXPORT Generator : public CodeGenerator {
                    std::string* error) const override;
 
   uint64_t GetSupportedFeatures() const override {
-    return FEATURE_PROTO3_OPTIONAL;
+    return Feature::FEATURE_PROTO3_OPTIONAL | Feature::FEATURE_SUPPORTS_EDITIONS;
```

Comment:
We probably should hold off on advertising support until it's ready.  Any protos under google/protobuf are allowlisted, and we could tweak php_proto_library to pass --experimental editions if that doesn't work

Reply:
Maybe I'm doing something wrong, but when I remove `FEATURE_SUPPORT_EDITIONS`, I get the following error:

```
test_messages_proto3_editions.proto: is an editions file, but code generator --php_out hasn't been updated to support editions yet.  Please ask the owner of this code generator to add support or switch back to proto2/proto3.
```

Reply:
protoc will skip this check for any proto path under "google/protobuf/" and "upb/".  We should probably add "editions." to [that list](https://github.com/protocolbuffers/protobuf/blob/d3b2fc52a4aec2dc9711c8235cbcdb71d4c10295/src/google/protobuf/compiler/code_generator.cc#L147) too

Reply:
Alternatively, you could modify `internal_php_proto_library` to accept an experimental_editions argument and have it set `--experimental_editions` on protoc

Reply:
I looked into both of these and, to do the first, `editions/` does not bypass the check because the filename being passed in is `test_messages_proto3_editions.proto` (the `editions/` path is not part of the filename, I am not sure why). As for the second, I cannot find a way to set custom flags on the `protoc` binary, so it seems like I'd have to modify `_proto_gen`, which I am not super excited about doing.

Reply:
Okay! Using the same flags added by Objective C in 6c7c5a5d2a6befdc4117242671612146efa50679, I removed the editions flag from PHP and added the `experimental_editions` flag for `protoc` (3b212ea2c0378a9f3ceace19d8acae7c5cf9898e)